### PR TITLE
ON HOLD: Add torrent info endpoint

### LINKED ISSUE #2586: Specific torrent info endpoint
- [x] Endpoint that exposes all information locally available about a specific torrent.
- [ ] Including the files with name and size, if available.
- [ ] Including if it is currently in the downloads list, if so: exposes all status information.

### PR DESCRIPTION
**Pull request by brussee**
Closes #2586
- [ ] Add flags for files & download status
- [ ] Cross match downloads with torrents by infohash

### CODE REVIEW

**Comment by brussee** [comment]
Win32 dirty reactor on 
Tribler.Test.Core.Modules.RestApi.Channels.test_channels_playlist_endpoint.TestChannelsPlaylistEndpoints.test_create_playlist_no_name
```
ERROR   1477344634            util:Tribler.dispersy.util:239  Exception raised on the reactor's thread AssertionError: "The reactor was dirty during tearDown".
 Traceback from this thread:
  File "C:\Python27\lib\runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "C:\Python27\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\Python27\Scripts\nosetests.exe\__main__.py", line 9, in <module>
    sys.exit(run_exit())
  File "C:\Python27\lib\site-packages\nose\core.py", line 121, in __init__
    **extra_args)
  File "C:\Python27\lib\unittest\main.py", line 95, in __init__
    self.runTests()
  File "C:\Python27\lib\site-packages\nose\core.py", line 207, in runTests
    result = self.testRunner.run(self.test)
  File "C:\Python27\lib\site-packages\nose\core.py", line 62, in run
    test(result)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 224, in run
    test(orig)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 224, in run
    test(orig)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 224, in run
    test(orig)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 224, in run
    test(orig)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 224, in run
    test(orig)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 224, in run
    test(orig)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 224, in run
    test(orig)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 224, in run
    test(orig)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 224, in run
    test(orig)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 224, in run
    test(orig)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 224, in run
    test(orig)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "C:\Python27\lib\site-packages\nose\suite.py", line 224, in run
    test(orig)
  File "C:\Python27\lib\site-packages\nose\case.py", line 45, in __call__
    return self.run(*arg, **kwarg)
  File "C:\Python27\lib\site-packages\nose\case.py", line 133, in run
    self.runTest(result)
  File "C:\Python27\lib\site-packages\nose\case.py", line 151, in runTest
    test(result)
  File "C:\Python27\lib\unittest\case.py", line 393, in __call__
    return self.run(*args, **kwds)
  File "C:\Python27\lib\unittest\case.py", line 358, in run
    self.tearDown()
  File "C:\workspace\jenkins\workspace\GH_Tribler_PR_tests_win32\tribler\Tribler\dispersy\util.py", line 46, in helper
    return blockingCallFromThread(reactor, func, *args, **kargs)
  File "C:\workspace\jenkins\workspace\GH_Tribler_PR_tests_win32\tribler\Tribler\dispersy\util.py", line 236, in blockingCallFromThread
    this_thread_tb = traceback.extract_stack()

 Traceback from the reactor's thread:
   File "C:\Python27\lib\site-packages\twisted\internet\defer.py", line 1128, in _inlineCallbacks
    result = g.send(result)
  File "C:\workspace\jenkins\workspace\GH_Tribler_PR_tests_win32\tribler\Tribler\Test\test_as_server.py", line 140, in checkReactor
    self.assertFalse(delayed_calls, "The reactor was dirty during %s" % phase)
  File "C:\Python27\lib\unittest\case.py", line 416, in assertFalse
    raise self.failureException(msg)
```

**Comment by brussee** [comment]
retest this please

File: Tribler/Core/Modules/restapi/channels/channels_torrents_endpoint.py [hunk 1]
```diff
@@ -218,6 +218,49 @@ def _on_add_failed(failure):
         self.deferred.addErrback(_on_add_failed)
         return NOT_DONE_YET
 
+    def render_GET(self, request):
+        """
+        .. http:get:: /channels/discovered/(string: channelid)/torrents/(string: torrent infohash)
+
+        Get information of torrent with a given infohash from a given channel.
```
Comment (devos50) [inline Tribler/Core/Modules/restapi/channels/channels_torrents_endpoint.py]: 'of a torrent' ...

File: Tribler/Test/Core/Modules/RestApi/Channels/test_channels_torrents_endpoint.py [hunk 2]
```diff
@@ -252,25 +252,76 @@ def verify_error_message(body):
...
+        url = 'channels/discovered/%s/torrents/abcd' % 'fakedispersyid'.encode('hex')
+        return self.do_request(url, expected_code=404)
+
+    @deferred(timeout=10)
+    def test_info_torrent(self):
+        """
+        Testing whether the API returns the right information for a request of a specific torrent from a channel
+        """
+        mock_channel_community = MockObject()
+        mock_channel_community.called_remove = False
```
Comment (devos50) [inline Tribler/Test/Core/Modules/RestApi/Channels/test_channels_torrents_endpoint.py]: Why are you setting the `called_remove` attribute to `False`?

**Comment by brussee** [comment]
@devos50 Done processing your requested changes.

**Comment by brussee** [comment]
TODO:
- [ ] Add flags for files & download status
- [ ] Cross match downloads with torrents by infohash

**Comment by brussee** [comment]
This is not wip (for me anymore).

**Comment by devos50** [comment]
Ok, since this is not needed for the desktop client for now, I will close this PR. If anyone wants to continue working on it, feel free to fork it and open a new PR.

**Comment by brussee** [comment]
@devos50 This PR is ready to merge, because someone can add the 2 enhancements later.

**Comment by devos50** [comment]
Oh, I thought it was not finished yet. In that case, I will review it later.

**Comment by brussee** [comment]
@devos50 I mean: if you split the description and issue #2586.

**Comment by devos50** [comment]
Will be included in #3219

**Pull request closed without merging**